# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.h
# Commit: 02269004c755
# Full Hash: 02269004c7554e6972c13ee6d147fc2ddcee34f7
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-26 06:07 +0000
# Regressor Bug: 1968890
# File Overlap Count: 5
# Overlapping Files: js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, js/loader/ModuleLoadRequest.h, js/loader/ModuleLoaderBase.h, dom/script/ScriptLoader.cpp
# Description:
#   Bug 1968890 - Part 7: Add Cancel method. r=jonco
# ==============================================================================

diff -r 9e2a3327cd0f -r 02269004c755 js/loader/ModuleLoaderBase.h
--- a/js/loader/ModuleLoaderBase.h	Fri Sep 26 06:07:44 2025 +0000
+++ b/js/loader/ModuleLoaderBase.h	Fri Sep 26 06:07:44 2025 +0000
@@ -574,6 +574,7 @@
 
   void OnFetchSucceeded(ModuleLoadRequest* aRequest);
   void OnFetchFailed(ModuleLoadRequest* aRequest);
+  void Cancel(ModuleLoadRequest* aRequest);
 
   // The slot stored in ImportMetaResolve function.
   enum class ImportMetaSlots : uint32_t { ModulePrivateSlot = 0, SlotCount };
